tests/extensions/test_extensions.py

```diff
@@ -130,3 +130,19 @@ def do(self, *args):
     comp.do.assert_called_once_with('before_training')
     a.do.assert_called_once_with('after_training')
     b.do.assert_called_once_with('after_batch')
+
+
+def test_simple_extension_before_batch_callback():
+
+    class Foo(SimpleExtension):
+        def __init__(self, **kwargs):
+            self.do = Mock()
+            super(Foo, self).__init__(**kwargs)
+
+        def do(self, which_callback, *args):
```

Comment:
Oh, crap, I missed this one.

On 5 July 2016 at 12:47, Philemon Brakel <notifications@github.com> wrote:

> In tests/extensions/test_extensions.py
> <https://github.com/mila-udem/blocks/pull/1126#discussion_r69625636>:
>
> > @@ -130,3 +130,19 @@ def do(self, *args):
> >      comp.do.assert_called_once_with('before_training')
> >      a.do.assert_called_once_with('after_training')
> >      b.do.assert_called_once_with('after_batch')
> > +
> > +
> > +def test_simple_extension_before_batch_callback():
> > +
> > +    class Foo(SimpleExtension):
> > +        def __init__(self, **kwargs):
> > +            self.do = Mock()
> > +            super(Foo, self).__init__(**kwargs)
> > +
> > +        def do(self, which_callback, *args):
>
> You're right. I actually copied that part from another test and it seems
> there are a couple of locations in this file where this happens. Again,
> also for future refactoring :-).
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/mila-udem/blocks/pull/1126/files/3f32dd1a522355e38d9d8ade2a76776eca93df6c#r69625636>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAn8YpsGD77s-rahFTabOCGKT2UZpcPsks5qSrTsgaJpZM4JFYTf>
> .
>